litmus-portal/backend/workflow-agent/pkg/cluster/events/util.go

```diff
@@ -0,0 +1,76 @@
+package events
+
+import (
+	v1alpha13 "github.com/argoproj/argo/pkg/apis/workflow/v1alpha1"
+	"github.com/gdsoumya/workflow_manager/pkg/types"
+	"github.com/litmuschaos/chaos-operator/pkg/apis/litmuschaos/v1alpha1"
+	v1alpha12 "github.com/litmuschaos/chaos-operator/pkg/client/clientset/versioned/typed/litmuschaos/v1alpha1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"strconv"
+)
+
+// util function, extracts the chaos data using the litmus go-client
+func getChaosData(engineName, engineNS string, chaosClient *v1alpha12.LitmuschaosV1alpha1Client) (*types.ChaosData, error) {
+	cd := &types.ChaosData{}
+	crd := &v1alpha1.ChaosEngine{}
+	cd.EngineName = engineName
+	cd.Namespace = engineNS
+	var err error
+	if cd.Namespace != "" {
+		crd, err = chaosClient.ChaosEngines(cd.Namespace).Get(cd.EngineName, v1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		crd, err = chaosClient.ChaosEngines("").Get(cd.EngineName, v1.GetOptions{})
```

Comment:
Why empty string in chaosEngines()?

Reply:
That is for cluster wide access, according to examples putting "" makes it cluster wide instead of namespace specific.